pkg/labels/labels.go

```diff
@@ -143,6 +147,10 @@ type Label struct {
 	//
 	// +kubebuilder:validation:Optional
 	Source string `json:"source"`
+
+	// optimization for CIDR prefixes
+	// +deepequal-gen=false
+	cidr *netip.Prefix `json:"-"`
```

Comment:
Should we lazy eval here and turn `cidr` into a private method rather than a member to always be pre-calculated. The cidr label will not always be compared or parsed. Thoughts?

Reply:
I looked in to this, and it is probably better not to lazy eval. Every time we add a CIDR identity, we do (pseudocode)

```
for selector in selectors: (N = 100)
  for identity in identities: (N = 1000)
    for label in identity: (N = 2)
      if label.Has(Selector):
        // ...
```

means there is definitely a cost to computing the CIDR every time. Plus, doing so requires heap allocation, since there are string substitutions. Meaning we would throw off a lot of garbage.